magento/pwa-studio
# Webpack client side caching.

## Description
This PR addresses concerns and solutions raised in #1623. 

## Related Issue
Closes #1623.

## Verification Steps
1. Build the venia code base using `yarn build`.
2. Stage the newly created build using `yarn stage:venia`.
3. Open a browser tab with cache cleared or try using an incognito tab.
4. Open network dev tools tab.
5. Now go to the venia instance using the URL that was provided in the terminal during stage process.
6. The first time you should see `client.[hash].js`, `runtime.[hash].js`, `vendors.[hash].js`, `registerSW.[hash].js` and `sw.[hash].js` downloaded apart from other files that are not of significance in this PR.
7. When you refresh the UI again, no files have to redownload. Meaning they have been cached the first time.
8. Now stop the staging server and run steps 1, 2 and 5 again. This time again since the code has not changed the build should not change and nothing should be downloaded from the server.
9. Again stop the staging server, try changing couple lines of code and redo steps 1, 2 and 5. This time only the client.[hash].js should be downloaded because the app build has changed.

## Screenshots / Screen Captures (if appropriate)
Check out #1623 for screenshots and metrics.

## Checklist
* App logic should not have changed. These are code build changes.
